tapgarden/planter.go

```diff
@@ -1698,6 +1774,38 @@ func (c *ChainPlanter) fundBatch(ctx context.Context, params FundParams,
 	return nil
 }
 
+// matchPsbtToGroupReq attempts to match a signed group virtual PSBT to a
+// corresponding group key request. This is done by comparing the tapscript root
+// in the PSBT to the tapscript root in the group key request.
+func matchPsbtToGroupReq(psbt psbt.Packet,
```

Comment:
Could we perhaps do the matching of this on the `rpcserver` level at least? Then the planter wouldn't need any new logic and the new field vs. deprecated field would only be visible on the RPC level?
I think we can get the current group key requests in the RPC server by querying the pending batch. Or am I missing something?

Reply:
I think I looked into that approach but the pending batch didn't include everything I needed. I think it may not be verbose? I can look into this again.

Reply:
With the latest version the matching seems to be quite slim and contained to this file, so don't spend too much time on it.